mqtt/mqtt-edgehub/src/auth/authorization/edgehub.rs

```diff
@@ -176,17 +234,24 @@ impl Authorizer for EdgeHubAuthorizer {
         Ok(auth)
     }
 
-    fn update(&self, update: Box<dyn Any>) -> Result<(), Self::Error> {
+    fn update(&mut self, update: Box<dyn Any>) -> Result<(), Self::Error> {
         let identities = update.as_ref();
-        if let Some(service_identities) = identities.downcast_ref::<ServiceIdentity>() {
-            debug!("{:?}", service_identities);
-            // TODO: fill in update method
+        if let Some(service_identities) = identities.downcast_ref::<Vec<ServiceIdentity>>() {
+            debug!(
+                "service identities update received. Service identities: {:?}",
+                service_identities
+            );
+
+            self.service_identities_cache = service_identities
```

Comment:
I believe we can consume update here and save some clones
```suggestion
            self.service_identities_cache = service_identities
                 .into_iter()
                .map(|id| (d.identity().into(), id))
                .collect();
```